spec/controllers/api/v1/admin/canonical_email_blocks_controller_spec.rb

```diff
@@ -5,23 +5,183 @@
 describe Api::V1::Admin::CanonicalEmailBlocksController do
   render_views
 
-  let(:user)    { Fabricate(:user, role: UserRole.find_by(name: 'Admin')) }
-  let(:token)   { Fabricate(:accessible_access_token, resource_owner_id: user.id, scopes: 'admin:read') }
+  let(:role)    { UserRole.find_by(name: 'Admin') }
+  let(:user)    { Fabricate(:user, role: role) }
+  let(:token)   { Fabricate(:accessible_access_token, resource_owner_id: user.id, scopes: scopes) }
   let(:account) { Fabricate(:account) }
+  let(:scopes)  { 'admin:read:canonical_email_blocks admin:write:canonical_email_blocks' }
 
   before do
     allow(controller).to receive(:doorkeeper_token) { token }
   end
 
+  shared_examples 'forbidden for wrong scope' do |wrong_scope|
+    let(:scopes) { wrong_scope }
+
+    it 'returns http forbidden' do
+      expect(response).to have_http_status(403)
+    end
+  end
+
+  shared_examples 'forbidden for wrong role' do |wrong_role|
+    let(:role) { UserRole.find_by(name: wrong_role) }
+
+    it 'returns http forbidden' do
+      expect(response).to have_http_status(403)
+    end
+  end
+
   describe 'GET #index' do
+    context 'with wrong scope' do
+      before do
+        get :index
+      end
+
+      it_behaves_like 'forbidden for wrong scope', 'read:statuses'
```

Comment:
Instead of using a before hook, I found this could be done by defining variables with `let` inside a block passed to `it_behaves_like`, as follows:
```rb
it_behaves_like 'forbidden for wrong scope' do
  let(:scopes) { 'read:statuses' }
  let(:action) { get :index }
end 
```
Then, the `shared_examples` could be written as follows:
```rb
shared_examples 'forbidden for wrong scope' do
  it 'returns http forbidden' do
    action
      
    expect(response).to have_http_status(403)
  end
end
```
I don't know which one preferable.